src/test/java/co/rsk/federate/signing/hsm/message/AdvanceBlockchainMessageTest.java

```diff
@@ -9,39 +13,77 @@
 import java.util.Collections;
 import java.util.List;
 
-import static org.mockito.Mockito.mock;
-import static org.mockito.Mockito.when;
-
 public class AdvanceBlockchainMessageTest {
 
     @Test
-    public void getBlockHeaders_ok() {
-        byte[] encodedBlockHeader1 = new byte[]{ 1 };
-        BlockHeader blockHeader1 = mock(BlockHeader.class);
-        when(blockHeader1.getFullEncoded()).thenReturn(encodedBlockHeader1);
+    public void test_getParsedBlockHeaders_ok() {
+        BlockHeader blockHeader1 = TestUtils.createBlockHeaderMock(1);
+        BlockHeader blockHeader2 = TestUtils.createBlockHeaderMock(2);
+        BlockHeader blockHeader4 = TestUtils.createBlockHeaderMock(4);
+
+        List<Block> blocks = Arrays.asList(
+            TestUtils.mockBlockWithBrothers(1, TestUtils.createHash(1), Arrays.asList(blockHeader1, blockHeader2)),
+            TestUtils.mockBlockWithBrothers(2, TestUtils.createHash(2), Collections.emptyList()),
+            TestUtils.mockBlockWithBrothers(3, TestUtils.createHash(3), Collections.singletonList(blockHeader4)));
 
-        List<BlockHeader> blockHeaders = Collections.singletonList(blockHeader1);
-        AdvanceBlockchainMessage message = new AdvanceBlockchainMessage(blockHeaders);
+        AdvanceBlockchainMessage message = new AdvanceBlockchainMessage(blocks);
 
-        Assert.assertEquals(1, message.getBlockHeaders().size());
-        Assert.assertEquals(Hex.toHexString(encodedBlockHeader1), message.getBlockHeaders().get(0));
+        Assert.assertEquals(3, message.getParsedBlockHeaders().size());
```

Comment:
This test and the one below are basically the same, right? This one just has less assertions

Reply:
The Test Above Asserts That The Size Is Correct, The One Below Asserts They Are In The Sorted Order

Reply:
That sounds like it could be the same test. One test with 2 assertions, no need to repeat it

Reply:
Updated